src/ray/core_worker/core_worker.cc

```diff
@@ -384,6 +384,7 @@ Status CoreWorker::Put(const RayObject &object, ObjectID *object_id) {
                                 worker_context_.GetNextPutIndex(),
                                 static_cast<uint8_t>(TaskTransportType::RAYLET));
   reference_counter_->AddOwnedObject(*object_id, GetCallerId(), rpc_address_);
+  RAY_CHECK_OK(local_raylet_client_->PinObjectIDs(rpc_address_, {*object_id}));
```

Comment:
Don't we need to reverse this line and the `Put`? Or else the raylet could try to get the object IDs before they are available in plasma.

Reply:
Good catch!